ui/src/utils/axios.ts

```diff
@@ -8,7 +15,33 @@ const instance = axios.create({
 });
 
 instance.interceptors.response.use(
-  (response) => response,
+  (response) => {
+    const env = process.env.NODE_ENV;
+    // If development, mock api calls
+    if (env === 'development' && response.config.method === 'get') {
+      const url = response.config.url;
+      if (url === '/services/') {
+        response.data = services;
+      } else if (url === '/server/') {
+        response.data = serverApps;
+      } else if (url?.match(/^\/server\/.*$/)) {
+        const serverApp = serverApps.user_apps.find(
+          (app) => app.name === url.split('/')[2],
+        );
+        if (serverApp) {
+          response.data = serverApp;
+        }
+      } else if (url === '/frameworks/') {
+        response.data = frameworks;
+      } else if (url === '/conda-environments/') {
+        response.data = environments;
+      } else if (url === '/spawner-profiles/') {
+        response.data = profiles;
+      }
```

Comment:
for most of these if(s) here, how do you feel about creating a map above and using that?

like say:

```javascript
const urlPathResponseDataMap = {
    "/services/": services,
    "/frameworks/": frameworks,
    "/conda-environments/": environments,
    ....
}
```

Reply:
Good call. Done